File: rails/app/controllers/api/v1/fizzbuzz_controller.rb
class Api::V1::FizzbuzzController < ApplicationController
  def index
    new_counter_value = Redis.current.incr(redis_fizzbuzz_key)
    render json: {
      value: FizzBuzzer.new.get_fizzbuzzed_string(new_counter_value),
      app_name: Rails.application.class.module_parent_name,
      app_version: File.read(Rails.root.join("version")).chomp,
    }
  end

  def reset
    Redis.current.del(redis_fizzbuzz_key)
  end

  private

  def redis_fizzbuzz_key
    ENV["FIZZBUZZ_REDIS_COUNTER_KEY"] || "fizzbuzz-counter"
  end
end

File: nodejs/controllers/fizzbuzz-controller.js
const FizzBuzzer = require('../services/fizzbuzzer');

module.exports = class FizzBuzzController {
  constructor(){
    this._fizzbuzzer = new FizzBuzzer();
    this._redis = require('../libs/redis');
  }

  async get(_, res){
    const newCounterValue = await this._redis.incr(this.redisCounterKey);
    const fizzbuzzedString = this._fizzbuzzer.getFizzBuzzedString(newCounterValue).toString();
    return res.json({
      value: fizzbuzzedString,
      app_name: process.env.npm_package_name,
      app_version: process.env.npm_package_version
    });
  }

  async delete(_, res){
    await this._redis.del(this.redisCounterKey);
    return res.status(204).send();
  }

  get redisCounterKey(){
    return process.env.FIZZBUZZ_REDIS_COUNTER_KEY || 'fizzbuzz-counter';
  }
}